### PR TITLE
[FLINK-20093] Create a download page for all optional sql client components

### PR DESCRIPTION
**Pull request by dawidwys**
## What is the purpose of the change

The PR creates a single page that lists all optional binaries for sql client. In the end we could link that page from the "Optional components" section in the https://flink.apache.org/downloads.html

## Brief change log
  - Adds a `liquify.rb` script which allows substituting variables in liquid variables
  - Creates a common mechanism for creating a table with download links for connector/format specific pages
  - Creates a page with an overview of all available format/connectors

Some screenshots:

Download:

image

Elasitcsearch (multiple versions):

image

Json (built-in connector):

image

Orc (no versions):

image

## Verifying this change

1. Build the documentation, check pages for all connector, formats and the `Download` page
2. Repeat with `is_stable` set to true in `_config.yml`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: docs/_data/sql-connectors.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+avro:
```
Comment (sjwiesman) [inline docs/_data/sql-connectors.yml:1]: Missing license header

File: docs/_includes/sql-connector-download-table.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<p>In order to use the {{ include.connector.name }} {{ include.connector.category }} the following
```
Comment (sjwiesman) [inline docs/_includes/sql-connector-download-table.html:1]: missing license header

File: docs/_plugins/liquify.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module Jekyll
```
Comment (sjwiesman) [inline docs/_plugins/liquify.rb:1]: Missing license header. Also, can you add some docs on this filter. I realize it's trivial but we've had issues of no one understanding these parts of the docs build in the past.

**Review by sjwiesman — CHANGES_REQUESTED** [review]
Thanks for doing this, it seems much more maintainable. Just a few nits on licensing and doc comments.

File: docs/_data/sql-connectors.yml (hunk 1 above)
Comment (sjwiesman) [reply docs/_data/sql-connectors.yml:1]: Could you also add a comment explaining how to add a new format or connector. I want it to be really self-evident how to maintain this.

**Comment by dawidwys** [comment]
Thanks for the review @sjwiesman I tried addressing your comments.

**Review by sjwiesman — APPROVED** [review]
+1. Please either apply your changes to the 'zh' files or create a follow-up ticket if those pages have already been translated.

**Comment by dawidwys** [comment]
I created https://issues.apache.org/jira/browse/FLINK-20129

**Comment by sjwiesman** [comment]
Merging this so #14017 and #14049 can use it .

**Pull request closed without merging**